### PR TITLE
Add the tests

### PR DESCRIPTION
**Pull request by FelixE91**
- test precip type calculation
- test the full function with correct shapes and NaNs
- update variable names ( #4 )
- simplify and clean the diagnostic_prtype function
- homogenize the input: all as arrays and corresponding meta data

### CODE REVIEW

File: tests/test_calculate_helper_functions.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    precipitation_intensity_grid_mmph = np.ones((3, 4)) # precipitation of 1 mm/h
+    snow_level_grid_m = np.ones((3, 4)) * 100 # snow level at 100 m
+    topography_grid_m = np.ones((3, 4)) * 300 # topography at 300 m
+    ground_temperature_grid_degC = np.ones((3, 4)) * -10
+    temperature_grid_degC = np.ones((3, 4)) * -5
+    result = calculate_precip_type(snow_level_grid_m, temperature_grid_degC, ground_temperature_grid_degC, precipitation_intensity_grid_mmph, topography_grid_m)
+    expected_result = np.ones((3, 4)) * 3
+    assert_array_almost_equal(result, expected_result)
+
+### The following will never give snow since topography < snow level: should be freezing rain case T2m < TTO and TG < TG0
```
Comment (ladc) [inline tests/test_calculate_helper_functions.py:46]: So this test has been removed because the expected output has been changed due to the change implemented as suggested by Maarten, am I correct?

**Review by ladc — APPROVED** [review]
Sorry for the extremely long wait. This looks fine. Thanks for looking into more detail into the edge case with the snow conditions. I suppose the new version implements Maarten's newly suggested approach?

File: tests/test_calculate_helper_functions.py (hunk 1 above)
Comment (FelixE91) [reply tests/test_calculate_helper_functions.py:46]: No, that has nothing to do with the proposed change to the INCA precipitation type calculation. That would simply never lead to the expected results. If the topography is below the snow line, it is never snow
(see here: https://github.com/pySTEPS/pysteps-diagnostic-prtype/blob/75d8336fb254cbc486c2c4e0718f97df373c7c42/pysteps_diagnostic_prtype/diagnostic/diagnostic_prtype.py#L520 ; with topoZSDiffGrid > 0).
I think we can delete the commented test to avoid confusion. It is simply wrong.

**Comment by FelixE91** [comment]
This is still using the current INCA algorithm without the suggested modification. I opened #6
